src/libAtomVM/opcodesswitch.h

```diff
@@ -108,14 +108,17 @@ typedef struct
 
 typedef dreg_t dreg_gc_safe_t;
 
+#define DEST_REGISTER(reg) dreg_t reg
+#define GC_SAFE_DEST_REGISTER(reg) dreg_gc_safe_t reg
+
 #define T_DEST_REG(dreg) \
     reg_type_c((dreg).reg_type), (int) ((dreg).index)
 
 #define T_DEST_REG_GC_SAFE(dreg) T_DEST_REG(dreg)
 #else
 
-typedef struct {} dreg_t;
-typedef dreg_t dreg_gc_safe_t;
+#define DEST_REGISTER(...)
+#define GC_SAFE_DEST_REGISTER(...)
```

Comment:
So just so I understand, the macro is empty if we are doing code loading but not if code loading and trace is enabled, is that right?  (Below, it looks like we are allocating on the stack when running the emulator)

Reply:
right.